pkg/network/ebpf/c/protocols/kafka/kafka-classification.h

```diff
@@ -268,6 +268,44 @@ static __always_inline bool validate_first_topic_name(pktbuf_t pkt, bool flexibl
     CHECK_STRING_VALID_TOPIC_NAME(TOPIC_NAME_MAX_STRING_SIZE_TO_VALIDATE, topic_name_size, topic_name);
 }
 
+// Reads the first topic id (can be multiple) from the given offset,
+// verifies if it is a valid UUID version 4
+static __always_inline bool validate_first_topic_id(pktbuf_t pkt, bool flexible, u32 offset) {
+    // The topic id is a UUID, which is 16 bytes long.
+    // It is in network byte order (big-endian)
+    u8 topic_id[16] = {};
+
+    // Skipping number of entries for now
+    if (flexible) {
+        if (!skip_varint_number_of_topics(pkt, &offset)) {
+            return false;
+        }
+    } else {
+        offset += sizeof(s32);
+    }
+
+    if (offset + sizeof(topic_id) > pktbuf_data_end(pkt)) {
+        return false;
+    }
+
+    pktbuf_load_bytes(pkt, offset, topic_id, sizeof(topic_id));
```

Comment:
we should be using the `_with_telemetry` if that's possible

Reply:
Should we always call the `_with_telemetry` variation? because we have other places in this file without it

Reply:
yes we should
In the past, we had places in which we had to avoid using `_with_telemetry` as we hit the instruction limit (or complexity limit). As long as we can use `_with_telemetry` we should